assets/javascripts/discourse-markdown/notifications.js.es6

```diff
@@ -9,43 +9,47 @@ export function setup(helper) {
             'div.p-notification--negative',
             'div.p-notification--important',
             'p.p-notification__response',
-            'span.p-notification__status',
-            'span.p-notification__line'
+            'span.p-notification__status'
         ]
     );
 
     helper.registerPlugin(md=>{
-        md.block.bbcode.ruler.push('note', {
-            tag: 'note',
+        md.block.bbcode.ruler.push("note", {
+            tag: "note",
             replace: function(state, tagInfo, content) {
-                let status = "";
-                let type = "";
-
-                if ('status' in tagInfo.attrs) {
-                    status = '<span class="p-notification__status">'
-                    + tagInfo.attrs.status
-                    + ':</span>'
-                }
-
+                // Build up the notification class from the specified "type"
+                // "p-notification--{type}"
+                let notificationClass = 'p-notification'
                 if (
                     'type' in tagInfo.attrs
                     && ['caution', 'positive', 'negative', 'important'].includes(
                         tagInfo.attrs.type.toLowerCase()
                     )
                 ) {
-                    type = "--" + tagInfo.attrs.type.toLowerCase();
+                    notificationClass += "--" + tagInfo.attrs.type.toLowerCase();
                 }
 
-                let notification_block = state.push('html_raw', '', 0);
+                // Start wrapper elements:
+                // <div class="p-notification"><p class="p-notification__render">
+                state.push('div_open', 'div', 1).attrSet('class', notificationClass);
+                state.push('paragraph_open', 'p', 1).attrSet('class', 'p-notification__response');
+
+                // Add status:
+                // <span class="p-notification__status">{status}</span>
+                if ('status' in tagInfo.attrs) {
+                    state.push('span_open', 'span', 1).attrSet('class', 'p-notification__status');
+                    state.push('text', '', 0).content  = tagInfo.attrs.status + ": ";
```

Comment:
I don't know enough about the surrounding code to know if .content takes markup or not, so please verify that it doesn't.

Reply:
This passes the "content" through the same parser as the rest of the post's Markdown. So while it does allow certain HTML tags (`<i>`, `<strong>`, `<a href>`) it strips malicious attributes and invalid `href` targets like `href="javascript:"`.

I've tested every case I can come up with, but @b-m-f, @anthonydillon please double-check this in your testing.

Reply:
Slight correction - I mistook the line you commented on. My description actually referred to 47, where we process the main plugin's `content`.

In this case, adding text to the `content` of a "text" node is even more safe - it renders the text exactly as-is. As in, `<i onclick=alert()>hello</i>` will be output as the text "<i onclick=alert()>hello</i>", completely unparsed.